From 31d8500b376b532db1f8111f822a91a2501b6f22 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Wed, 19 Feb 2025 11:01:02 +0100
Subject: [PATCH] openssh: Update to version 9.9p2

Update to the latest version of openssh. Contains several security
issues (CVE-2025-26465, CVE-2025-26466). Remove patch, that is
already part of upstream. For complete changes, see:

https://www.openssh.com/releasenotes.html#9.9p2
---
 net/openssh/Makefile                          |  4 +--
 ...-construct_utmp-when-USE_BTMP-is-set.patch | 30 -------------------
 2 files changed, 2 insertions(+), 32 deletions(-)
 delete mode 100644 net/openssh/patches/001-build-construct_utmp-when-USE_BTMP-is-set.patch

diff --git a/net/openssh/Makefile b/net/openssh/Makefile
index c86a431f0..e968609bf 100644
--- a/net/openssh/Makefile
+++ b/net/openssh/Makefile
@@ -8,8 +8,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=openssh
-PKG_REALVERSION:=9.9p1
-PKG_VERSION:=9.9_p1
+PKG_REALVERSION:=9.9p2
+PKG_VERSION:=9.9_p2
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_REALVERSION).tar.gz
diff --git a/net/openssh/patches/001-build-construct_utmp-when-USE_BTMP-is-set.patch b/net/openssh/patches/001-build-construct_utmp-when-USE_BTMP-is-set.patch
deleted file mode 100644
index 67a7062a4..000000000
--- a/net/openssh/patches/001-build-construct_utmp-when-USE_BTMP-is-set.patch
+++ /dev/null
@@ -1,30 +0,0 @@
-From 76a618d2842c34c16cd21a4efc7230e2f459008d Mon Sep 17 00:00:00 2001
-From: Damien Miller <djm@mindrot.org>
-Date: Wed, 25 Sep 2024 11:13:05 +1000
-Subject: [PATCH] build construct_utmp() when USE_BTMP is set
-
-Fixes compile error on Void Linux/Musl
----
- loginrec.c | 4 ++--
- 1 file changed, 2 insertions(+), 2 deletions(-)
-
---- a/loginrec.c
-+++ b/loginrec.c
-@@ -614,7 +614,7 @@ line_abbrevname(char *dst, const char *s
-  ** into account.
-  **/
- 
--#if defined(USE_UTMP) || defined (USE_WTMP) || defined (USE_LOGIN)
-+#if defined(USE_BTMP) || defined(USE_UTMP) || defined (USE_WTMP) || defined (USE_LOGIN)
- 
- /* build the utmp structure */
- void
-@@ -698,7 +698,7 @@ construct_utmp(struct logininfo *li,
- 	}
- # endif
- }
--#endif /* USE_UTMP || USE_WTMP || USE_LOGIN */
-+#endif /* USE_BTMP || USE_UTMP || USE_WTMP || USE_LOGIN */
- 
- /**
-  ** utmpx utility functions
-- 
2.48.1

